### DISCUSSION TITLE
Pypowerwall is stopped working after firmware upgrade - Tesla Pros screen?

### DISCUSSION THREAD

**Discussion by bparruck** [jasonacox/Powerwall-Dashboard#367] [General]
When I go to the backup gateway URL, I get this:
![image](https://github.com/jasonacox/Powerwall-Dashboard/assets/3960960/16886935-88c5-4387-a3c6-b571865cfbae)

**Comment by bparruck** [comment]
Also, the password reset is no longer allowed:
![image](https://github.com/jasonacox/Powerwall-Dashboard/assets/3960960/af8a479b-909f-4dfa-aae4-44746b4f8619)

**Reply by jasonacox** [reply]
Hi @bparruck  - Thanks for opening this. I'm not seeing any issues, but wonder if this is a new change due to firmware (the animation looks different).

* What is your firmware reported in the backup gateway portal?  I'm on 23.28.1.
* What errors are you seeing from pypowerwall?  `docker logs pypowerwall` 
* Do you see anything on your dashboard (screenshot would be great).

**Reply by Craig-G-NZ** [reply]
my powerwall must have just updated itself and i've started getting the same messages when i visit the powerwalls own internal website.

my pypowerwall is showing this now

![image](https://github.com/jasonacox/Powerwall-Dashboard/assets/92700720/08091e50-549b-4ef8-8e2f-0c618c4b8728)

**Reply by bparruck** [reply]
@Craig-G-NZ Are you still having issues connecting? Here's how it worked for me:

1. Turn off the PyPowerwall from accessing the backup gateway. I used: `./compose-dash.sh stop`
2. Log into the backup gateway portal to make sure the login (email/password) still works. This was difficult for me. It included a reset for the backup gateway using a pencil and using default password to setup the password used earlier.
3. Turn on the PyPowerwall from accessing the backup gateway. I used: `./compose-dash.sh start`

**Reply by Craig-G-NZ** [reply]
i havent doing anything specifically, but i think i have restarted my pypowerwall during a shutdown and restart of the server while replacing a disk. it seems to be working again now all good.

Its been a while since i've had a firmware update like that and forgot i just needed to restart pypowerwall when these things happen doh

**Comment by BJReplay** [comment]
> * What is your firmware reported in the backup gateway portal?  I'm on 23.28.1.

Also not currently seeing this on 23.28.2

I did see that message (the Tesla Pros message) at one point, but I think if you hit Continue, you end up at the login screen as usual.  I've never seen it since.

**Reply by bparruck** [reply]
Good to know that it is still working for you guys. I could not do the password last night and this morning it is no longer on the WiFi. I'll put it back on WiFi and figure this out.

I think they must have done a firmware upgrade. The firmware was 23.12.11 before the failure last night.

**Reply by bparruck** [reply]
Unfortunately, I need to figure out how to put the backup gateway on the WiFi, since the App is no longer seeing my WiFi network. Once I figure that out (with help from Tesla) and will update you guys.

Thanks for the quick response - it is good to know its only my system.

**Reply by jasonacox** [reply]
Scan your WiFi network to make sure it didn't just move to a new IP.  You can use:

```bash
pip install pypowerwall # if you haven't already

python -m pypowerwall scan
```

One thing to note: If somehow your password did change, if you are still running the pypowerwall container, it will keep trying to log in and could lock you out. Stop the container with `docker stop pypowerwall`.  It would also be good to see what errors it is throwing to verify if it sees a password issue or is just unable to find the Powerwall.

**Comment by bparruck** [comment]
I did manage to log back into the backup gateway after stopping the pypowerwall. I could restart the containers and everything is working. Nothing is broken with this new version of firmware.

**Reply by jasonacox** [reply]
That's great news!  So for some reason it did require that you stop pypowerwall to log in?

**Comment by bparruck** [comment]
@jasonacox When I was trying to access the backup gateway I kept getting error messages saying "too many API requests" and the portal would become unresponsive. Once I shutdown the containers (and its polling) it became more stable.

**Reply by jasonacox** [reply]
>"too many API requests"

Thanks @bparruck !  That's what I was wondering. Hopefully this thread will help anyone else who sees something similar.  Thanks for opening this issue.

**Comment by bparruck** [comment]
Thanks for responding quickly - @jasonacox @BJReplay. I was worried Tesla changed their API and broke this neat project.

**Comment by jasonacox** [comment]
Seeing other people report problems and wondering if their setup is broken. I'll reopen this to help others. 

Here is what is being reported. After a recent upgrade (23.28.x), if you log in to the Tesla Powerwall gateway portal you will see this:

<img width="657" alt="image" src="https://github.com/jasonacox/Powerwall-Dashboard/assets/836718/9a15769d-6a8f-4af9-ad5d-458bf90ff4ac">

If you click "Continue" it will go to this screen:

<img width="587" alt="image" src="https://github.com/jasonacox/Powerwall-Dashboard/assets/836718/781ff449-cb5b-47db-a8b3-b8c84619e151">

If you click "Upgrade Later" it will go to the regular login screen:

<img width="447" alt="image" src="https://github.com/jasonacox/Powerwall-Dashboard/assets/836718/3b84b916-fd10-43ab-9c24-86264858b444">

After entering your credentials, it looks like normal:

<img width="675" alt="image" src="https://github.com/jasonacox/Powerwall-Dashboard/assets/836718/72921ebc-19b6-4899-8226-a6409624d2c1">

---

Key points:

* You can bypass the Tesla Pro screen - hit "Continue" and "Upgrade Later", you end up at the login screen as usual.
* Do a hard refresh (clear browser cache) on your Dashboard browser or try a different browser (e.g. Chrome, Safari, Firefox).
* If your pypowerwall logs are showing errors - stop it for a while and restart to make sure it can log in again:`docker pypowerwall restart`
* You can scan to make sure your Powerwall didn't land on a different IP address after firmware upgrade using something like this: `python3 -m pypowerwall scan`
* Check to see if the proxy is giving a response: http://localhost:8675/version and http://localhost:8675/vitals (change localhost to the address of the machine running the proxy/Dashboard).
* From what I can tell, everything is the same, just that Tesla created an Android app for installers to use instead of having to use the web page setup. I do hope they don't remove the web based version.

**Reply by jasonacox** [reply]
Some people are reporting that "data collection has stopped working" after this upgrade and the Tesla Pros screen appeared. This didn't happen to me but enough people are reporting it that I would like to track and try to figure out what is happening. If this is happening to you, can you post your `pypowerwall` logs that might show any error? 

```bash
docker logs pypowerwall

# for some the fix is a restart but only if you see errors 
docker stop pypowerwall # wait a few minutes
docker start pypowerwall
```

**Reply by jeffkruczynski** [reply]
@jasonacox I am seeing this behavior now. I went through the following this evening:
- Upgraded from 2.10.0 to 3.0.2. That seemed to go fine, but the Power Flow pane was always updating, and clearly no new data were being collected.
- I found this thread and tried accessing the gateway (which is at 192.168.12.209 on my network). I had to cycle the switch on the Powerwall in order to successfully access, and got through the upgrade screens you showed above.
- Once I was logged in, the Powerwall Dashboard Power Flow pane started reflecting current state, and data collection had started again.
- After a few minutes, the pane stopped updating again, and I was not able to access the gateway. After some persistent trying, the gateway has returned, and the Power Flow pane is updating now, and data collection is happening.

I don't know why the gateway stopped providing info (I'm on 23.28.2 27626f98 by the way) but I'll continue to watch it to see whether it stops again.

**Reply by jasonacox** [reply]
> I had to cycle the switch on the Powerwall in order to successfully access

This would be needed for the "installer" access rather than the "customer" access method. I know the installer access times out and won't renew without a switch cycle as you mention. I suggest you make sure you follow the [instructions above](https://github.com/jasonacox/Powerwall-Dashboard/discussions/367#discussioncomment-7278760) to dismiss the Tesla Pros screen if it is still showing up (but use the customer login). And follow @bparruck [instructions below](https://github.com/jasonacox/Powerwall-Dashboard/discussions/367#discussioncomment-7282291) and see if that help. You should not need to cycle the switch.

**Comment by billraff** [comment]
I was seeing errors in my docker log that the Powerwall could not be found at the ip address that I had configured. I used that ip address in my browser to login to the gateway (after telling to upgrade later) and can see the normal screen there showing the status of the power walls. However, I sill am not receiving data in my docker pypowerall. I see this for the status -
<img width="509" alt="Screenshot 2023-10-14 at 12 55 38 PM" src="https://github.com/jasonacox/Powerwall-Dashboard/assets/29167787/d7edb817-bcfd-44bc-bd63-c29500980f90">
I did the restart at 12:21 and after that I don't see any errors though as seen in the first screenshot I am not seeing any data still.
<img width="1370" alt="Screenshot 2023-10-14 at 12 57 17 PM" src="https://github.com/jasonacox/Powerwall-Dashboard/assets/29167787/23cbf9eb-beba-4ba8-b863-0b3a06b3460f">
I'm running all this on my Sinology NAS.

**Reply by billraff** [reply]
Well, seems now I am starting to see data though Power Flow is still showing as in my first screen shot.

**Reply by jasonacox** [reply]
The power flow animation seems to indicate that the Tesla Pros upgrade popup is still there (e.g. the CONTINUE button).  Try clicking "Continue" or pointing your browser to the Powerwall Gateway IP and hit "Continue" and "Upgrade Later".  Also try doing a hard refresh on your Grafana browsers to see if it clears.

**Reply by billraff** [reply]
Sorted by clearing browser cache. Thanks all!

**Reply by jasonacox** [reply]
Thanks @billraff - I added that to my `key points` guide above.

**Comment by bparruck** [comment]
Here's how it worked for me:
1. Turn off the PyPowerwall from accessing the backup gateway. I used:` ./compose-dash.sh stop`
2.  Log into the backup gateway portal to make sure the login (email/password) still works. This was difficult for me.
3. Turn on the PyPowerwall from accessing the backup gateway. I used:` ./compose-dash.sh start`
